master/getting-started/kubernetes/installation/hardway/install-node.md

```diff
@@ -0,0 +1,333 @@
+```
+
+The certificate presents the Common Name (CN) as `calico-felix`, which is what we configured Typha to accept
+in the last lab.
+
+Sign the Felix certificate with the CA we created earlier
+```
+openssl x509 -req -in felix.csr \
+                  -CA typhaca.crt \
+                  -CAkey typhaca.key \
+                  -CAcreateserial \
+                  -out felix.crt \
+                  -days 365
+```
+
+Store the Felx key and certificate in a Secret that Felix will access
+
+```
+kubectl create secret generic -n kube-system calico-felix-certs --from-file=felix.key --from-file=felix.crt
+```
+
+## Provision RBAC
+
+Create the ServiceAccount that the `calico-node` container will run as
+
+```
+kubectl create serviceaccount -n kube-system calico-node
+```
+
+Provision a ClusterRole with permissions to read and modify {{site.prodname}} data store objects
+
+```
+kubectl apply -f - <<EOF 
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: calico-node
+rules:
+  - apiGroups: [""]
+    resources:
+      - endpoints
+      - services
+    verbs:
+      # Used to discover service IPs for advertisement.
+      - watch
+      - list
+      # Used to discover Typhas.
+      - get
+  - apiGroups: [""]
+    resources:
+      - nodes/status
+    verbs:
+      # Needed for clearing NodeNetworkUnavailable flag.
+      - patch
+      # Calico stores some configuration information in node annotations.
+      - update
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - globalfelixconfigs
+      - felixconfigurations
+      - bgppeers
+      - globalbgpconfigs
+      - bgpconfigurations
+      - ippools
+      - ipamblocks
+      - globalnetworkpolicies
+      - globalnetworksets
+      - networkpolicies
+      - clusterinformations
+      - hostendpoints
+      - blockaffinities
+      - networksets
+    verbs:
+      - get
+      - list
+      - watch
+  # Calico must create and update some CRDs on startup.
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - ippools
+      - felixconfigurations
+      - clusterinformations
+    verbs:
+      - create
+      - update
+  # Calico stores some configuration information on the node.
+  - apiGroups: [""]
+    resources:
+      - nodes
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - ipamconfigs
+    verbs:
+      - get
+  # Block affinities must also be watchable by confd for route aggregation.
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - blockaffinities
+    verbs:
+      - watch
+EOF
+```
+
+Bind the ClusterRole to the `calico-node` ServiceAccount
+
+```
+kubectl create clusterrolebinding calico-node --clusterrole=calico-node --serviceaccount=kube-system:calico-node
+```
+
+## Install DaemonSet
+
+`calico-node` runs as a DaemonSet so that it is installed on every node in the cluster.
+
+Create the DaemonSet
+
+```
+kubectl apply -f - <<EOF 
```

Comment:
I'm worried that all of these inline resources will diverge from our templated manifests very quickly.

What do we need to do in order to make this documentation use those manifests? Presumably it mostly has to do with the TLS stuff, which we don't have by default in the manifests?

Reply:
Resolving since this is by design.